### PR TITLE
boards: m5stack_atoms3lite: Add support for M5Stack AtomS3 Lite

### PR DESCRIPTION
**Pull request by MrMarteng**
Add support for M5Stack AtomS3 Lite development board.

The AtomS3Lite is a smaller version of the AtomS3 that
features only a StatusLED and no LCD display.

### CODE REVIEW

**Comment by MrMarteng** [comment]
Any update, comment on that PR?

File: tests/lib/devicetree/devices/testcase.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ tests:
       - bl5340_dvk_cpuapp
       - bl5340_dvk_cpuapp_ns
       - m5stack_core2
+      - m5stack_atoms3lite
```
Comment (kartben) [inline tests/lib/devicetree/devices/testcase.yaml:18]: might be worth mentioning why in the commit message?

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/xtensa/m5stack_atoms3lite/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+if BOARD_M5STACK_ATOMS3LITE
+
+config BOARD
+	default "m5stack_atoms3lite"
+	depends on BOARD_M5STACK_ATOMS3LITE
+
+config ENTROPY_GENERATOR
+	default y
+
+config HEAP_MEM_POOL_SIZE
```
Comment (LucasTambor) [inline boards/xtensa/m5stack_atoms3lite/Kconfig.defconfig:14]: This config was "recently" changed to `HEAP_MEM_POOL_ADD_SIZE_BOARD`

File: boards/xtensa/m5stack_atoms3lite/support/openocd.cfg
Comment (LucasTambor) [inline boards/xtensa/m5stack_atoms3lite/support/openocd.cfg:1]: In the docs is said that debugging is not supported, do we need this file?

**Review by LucasTambor — CHANGES_REQUESTED** [review]

File: boards/xtensa/m5stack_atoms3lite/support/openocd.cfg
Comment (kartben) [reply boards/xtensa/m5stack_atoms3lite/support/openocd.cfg:1]: Ah good point, probably worth cleaning things up in Core2 and StickCPlus as well, @MrMarteng?

Comment (MrMarteng) [reply boards/xtensa/m5stack_atoms3lite/support/openocd.cfg:1]: Good catch. I removed it from this PR and an added a task to my todo list to clean up the other m5stack boards as well. Thanks

File: boards/xtensa/m5stack_atoms3lite/Kconfig.defconfig (hunk 2 above)
Comment (MrMarteng) [reply boards/xtensa/m5stack_atoms3lite/Kconfig.defconfig]: fixed in [bd019a9](https://github.com/zephyrproject-rtos/zephyr/commit/bd019a9cf65db130247357c3d9de3a8da0d7a504)

File: tests/lib/devicetree/devices/testcase.yaml (hunk 1 above)
Comment (MrMarteng) [reply tests/lib/devicetree/devices/testcase.yaml]: Actually, when reviewing I saw that this is not really needed. Removed it from the devicetree test in [bd019a9](https://github.com/zephyrproject-rtos/zephyr/commit/bd019a9cf65db130247357c3d9de3a8da0d7a504)

File: boards/xtensa/m5stack_atoms3lite/Kconfig.board [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# M5Stack AtomS3Lite board configuration
+# Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+# SPDX-License-Identifier: Apache-2.0
+
+# M5Stack AtomS3-Lite
+config BOARD_M5STACK_ATOMS3LITE
+	bool "M5Stack AtomS3Lite Development Board"
```
[the remark this answers is not in the document]
Comment (MrMarteng) [reply boards/xtensa/m5stack_atoms3lite/Kconfig.board]: You are absolutely right that a constant naming is important. I followed your recommendation.
Your input is always much apperciated and welcome. I realize that I still need to improve the diligence of my PRs. You certainly have better things to do than finding my spelling mistakes or typos ;-)

**Review by LucasTambor — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**